File: Data Structures & Algorithms/task-scheduling/submission-1.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        
        int[] count = new int[26];
        for(int i=0; i<tasks.length; i++) {
            count[tasks[i]-'A']++;
        }

        Arrays.sort(count);
        int maxf = count[25];
        int idle = (maxf - 1)*n;

        for(int i=24; i>=0; i--) {
            idle -= Math.min(count[i], maxf-1);
        }

        return Math.max(0, idle)+tasks.length;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        i=0
        j=len(heights)-1
        area,maxarea=0,0
        while i<j:
            area=min(heights[i], heights[j])*(j-i)

            maxarea=max(maxarea, area)

            if heights[i] > heights[j]:
                j-=1
            else:
                i+=1
        
        return maxarea